### DISCUSSION TITLE
How to use the reset from another pages.

### DISCUSSION THREAD

**Discussion by MarceeloFerreira** [SupremeTechnopriest/react-idle-timer#280] [Q&A] [answered]
Hello!

I'm using the react-idle-timer and I have an application with 4 pages, so if the user stay idle for a few minutes, a modal is show and the user can stay logged or can logout, on the function for stay logged I want to reset the timer, but I dont know how to do it, if someone can help me, i will appreciate.

I'm using crossTab and startOnMount with true for now.

For now I just refresh the page.

Thanks for now!

Regards

**Comment by SupremeTechnopriest** [comment, marked as answer]
When your user clicks the "stay logged in" button, just call the `start()`, `reset()` or `activate()` methods on your IdleTimer instance.

https://idletimer.dev/docs/getting-started/new#integrated-prompting

**Reply by MarceeloFerreira** [reply]
Hi, i try to do something like this 

![image](https://user-images.githubusercontent.com/73614064/187508585-0f5c6fea-3b7a-4c63-8d64-d5db9a689d64.png)

I try with reset too but when the modal show the first time and i click on Stay Logged in, should be open modal again after timeout, but notting happens,

**Reply by SupremeTechnopriest** [reply]
Think you can put it in a code sandbox for me?  Then I can take a look.

**Reply by MarceeloFerreira** [reply]
This is an examlple.

When timeout reach (5 seconds), the modal open and on the modal component i try to use `reset()` function, I put the timer and try to reset but without success, if you find a solution i appreciate.

If need something just let me know.

Thanks by now.

https://codesandbox.io/s/react-idle-timer-74ne0e?file=/src/App.tsx

**Reply by SupremeTechnopriest** [reply]
I see the problem.  You are creating a separate instance of IdleTimer inside your SessionDialog component.  You have two options.  Use the provider or pass the reset function down in a property.

Here are the docs for the provider: https://idletimer.dev/docs/api/idle-timer-provider
And here is an example of passing the reset function to the child component: 
https://codesandbox.io/s/react-idle-timer-forked-u6rrsf?file=/src/App.tsx

I also set it up to use the new prompt feature in v5.  Let me know if this solves your issue.

**Reply by MarceeloFerreira** [reply]
Hello @SupremeTechnopriest 

Thanks for the help, I found the problem and solved it.
In my code I didn't pass the setOpen to the child component, so the timer was being reset by `reset()` but this way the modal only opened 1 time.
So I passed the setOpen problem solved, and now it's working and great!

Thanks a lot!

Regards,
Marcelo

**Reply by JSDUNIYA** [reply]
Can you please show code example how did you pass that SetOpen for reset ?
